tests/xen/test_ring0.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import pytest
+import secrets
+import time
+from typing import Optional
+
+from lib.host import Host
+from lib.vm import VM
+
+# Requirements:
+# From --hosts parameter:
+# - host: XCP-ng host >= 8.2, reboots after test
+# From --vm parameter:
+# - A VM to import
+# Only XST tests with pass/fail results, and that don't crash the host were included.
+
+
+def do_execute_xst(host: Host, modname: str, testname: Optional[str] = None):
+    if testname is None:
+        testname = modname
+    host.ssh(f"modprobe xst_{modname}")
+    try:
+        host.ssh(f"echo 1 > /sys/kernel/debug/xst/{testname}/run")
+        host.ssh(f"grep -q 'status: pass' /sys/kernel/debug/xst/{testname}/results")
+    finally:
+        host.ssh(f"modprobe -r xst_{modname}", check=False)
+
+
+@pytest.fixture
+def host_without_livepatch_loaded(host: Host):
+    if host.ssh_with_result("lsmod | grep -wq livepatch_tester").returncode == 0:
+        pytest.fail("livepatch_tester already loaded, host needs reboot")
```

Comment:
The goal of the fixture is to get an error rather than a failure.  Even if `pytest.fail` achieves that it looks awkward, wouldn't a simple `assert` do the job?

Reply:
Looks like it can even be a good idea, as the `Failed` exception raised by `pytest.fail` does not derive from `Exception` and should not be silenced by error by `except Exception` statements.